### DISCUSSION TITLE
Recieving raw binary over UART

### DISCUSSION THREAD

**Discussion by robzed** [tabemann/zeptoforth#35] [Q&A] [answered]
Hi,

I have a Forth word that receives the length (2 bytes of binary) and that number of binary bytes of raw data, from UART1 on a Pico, using KEY? and KEY. It puts this data in a buffer.

A different Forth word then send this over UART2 TX with 2 >UART (after checking >UART?)

Do you see any problems caused by the break keys in the uart driver? (Was it ^T ???). Should I get the PC program to escape bytes before sending to the UART1 for input via KEY? (the recieve word can easily translate back to raw binary).

**Comment by robzed** [comment]
I guess from the README on https://github.com/tabemann/zeptoforth I need to at least escape ^C and ^T. Maybe it's better to escape all control characters, to avoid problems in future?

**Reply by tabemann** [reply, marked as answer]
Yes, data receipt on UART0 (on the RP2040 the UART's are numbered 0 and 1) automatically captures ^C and ^T as it is meant to be used as the console, so you would need to escape them. This is not true of UART1, which does not capture any bytes.
